tests/unittests/StatsExporterTest.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/StatsExporter.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace glow;
+
+using testing::ElementsAre;
+
+class MockStatsExporter : public StatsExporter {
+public:
+  MockStatsExporter() { StatsExporter::registerStatsExporter(this); }
+
+  ~MockStatsExporter() override {}
+
+  void addQuantileValue(llvm::StringRef key, double value) override {
+    quantiles[key].push_back(value);
+  }
+
+  void incrementCounter(llvm::StringRef key, int64_t value) override {
+    counters[key] += value;
```

Comment:
I am biased from using fb303 but I think this this is fine and better than enforcing the use of const char* or enums for keys because of the the flexibility it adds.

Reply:
Previously I've used codegen here to wrap these calls with static functions that can't get the key name wrong (ie moving the flexibility to compile time). That worked very well but it's quite a bit of extra code to write.

Reply:
Say we want to export a memory usage counter per-device (for example), this wouldn't be possible with static strings or codegen unless the number of devices will be available at compile time.

Reply:
Kind of on the fence here.  Since fb303 uses string keys I'm inclined to leave the interface as is.  I could see making `glow.devices_used.interpreter` etc into constants rather than literals though.